hw/ip/aes/model/crypto.c

```diff
@@ -32,6 +32,9 @@ int crypto_encrypt(unsigned char *output, const unsigned char *iv,
     return -1;
   }
 
+  // Disable padding - We only ever encrypt multiples of 16 (the block size).
+  EVP_CIPHER_CTX_set_padding(ctx, 0);
```

Comment:
I'm not sure if it is strictly necessary but I am slightly worried about someone copying and using this code with padding disabled in some other context, so maybe even a slightly more seriously worded comment is appropriate here, something like:
`// Disable padding - it is safe to do so here because we only ever encrypt multiples of 16 (the block size)`
But you could adjust the wording to your preference.

Reply:
Thanks @cdgori for your comment. I did not think that far, but it makes sense. I adapted the comments accordingly.